src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -271,6 +278,1245 @@ ep_rt_mono_os_environment_get_utf16 (ep_rt_env_array_utf16_t *env_array)
 #endif
 }
 
+// Rundown flags.
+#define RUNTIME_SKU_CORECLR 0x2
+
+#define METHOD_FLAGS_DYNAMIC_METHOD 0x1
+#define METHOD_FLAGS_GENERIC_METHOD 0x2
+#define METHOD_FLAGS_SHARED_GENERIC_METHOD 0x4
+#define METHOD_FLAGS_JITTED_METHOD 0x8
+#define METHOD_FLAGS_JITTED_HELPER_METHOD 0x10
+
+#define MODULE_FLAGS_NATIVE_MODULE 0x2
+#define MODULE_FLAGS_DYNAMIC_MODULE 0x4
+#define MODULE_FLAGS_MANIFEST_MODULE 0x8
+
+#define ASSEMBLY_FLAGS_DYNAMIC_ASSEMBLY 0x2
+#define ASSEMBLY_FLAGS_NATIVE_ASSEMBLY 0x4
+#define ASSEMBLY_FLAGS_COLLECTIBLE_ASSEMBLY 0x8
+
+#define DOMAIN_FLAGS_DEFAULT_DOMAIN 0x1
+#define DOMAIN_FLAGS_EXECUTABLE_DOMAIN 0x2
+
+// Rundown events.
+EventPipeProvider *EventPipeProviderDotNETRuntimeRundown = NULL;
+EventPipeEvent *EventPipeEventMethodDCEndVerbose_V1 = NULL;
+EventPipeEvent *EventPipeEventDCEndInit_V1 = NULL;
+EventPipeEvent *EventPipeEventDCEndComplete_V1 = NULL;
+EventPipeEvent *EventPipeEventMethodDCEndILToNativeMap = NULL;
+EventPipeEvent *EventPipeEventDomainModuleDCEnd_V1 = NULL;
+EventPipeEvent *EventPipeEventModuleDCEnd_V2 = NULL;
+EventPipeEvent *EventPipeEventAssemblyDCEnd_V1 = NULL;
+EventPipeEvent *EventPipeEventAppDomainDCEnd_V1 = NULL;
+EventPipeEvent *EventPipeEventRuntimeInformationDCStart = NULL;
+
+/*
+ * Forward declares of all static rundown functions.
+ */
+
+static
+bool
+resize_buffer (
+	uint8_t **buffer,
+	size_t *size,
+	size_t current_size,
+	size_t new_size,
+	bool *fixed_buffer);
+
+static
+bool
+write_buffer (
+	const uint8_t *value,
+	size_t value_size,
+	uint8_t **buffer,
+	size_t *offset,
+	size_t *size,
+	bool *fixed_buffer);
+
+static
+bool
+write_buffer_string_utf8_t (
+	const ep_char8_t *value,
+	uint8_t **buffer,
+	size_t *offset,
+	size_t *size,
+	bool *fixed_buffer);
+
+static
+bool
+write_runtime_info_dc_start (
```

Comment:
I believe the equivalent of these functions in coreclr are autogenerated at build time from a manifest.  Rather than manually curating these, could we re-use the scripts that auto-generate them for coreclr?  It would ensure that they are in sync on both ends.  Preferably, we might use the same generated code, but the generated code uses C++ features I believe, so we'd have to modify them.

Reply:
Yes that is the long term goal, but wanted to get rundown working first on Mono and then spend time on converting the script into something that can be consumed by both runtimes.

Reply:
That sounds like a decent plan.